README.md

```diff
@@ -349,7 +349,7 @@ mail.cc              #=> 'sam@test.lindsaar.net'
 mail.subject         #=> "This is the subject"
 mail.date.to_s       #=> '21 Nov 1997 09:55:06 -0600'
 mail.message_id      #=> '<4D6AA7EB.6490534@xxx.xxx>'
-mail.decoded         #=> 'This is the body of the email...
+mail.body.decoded    #=> 'This is the body of the email...
```

Comment:
`mail.decoded` is the way to go. It checks whether the message is plain text, an attachment, or non-multipart, and raises the exception you're seeing on multipart messages:
https://github.com/mikel/mail/blob/8471b01760196c7b510ddabce9c50f72c5e79a2f/lib/mail/message.rb#L1886-L1897

If you want to extract the text part of an email, if present:
```ruby
if mail.multipart?
  mail.text_part&.decoded
else
  mail.decoded
end
```

Reply:
Thanks for the explanation. If I'd had the time I would craft a PR to change the Exception (I probably did not look up what is happening because I saw the `NoMethodError`, alternatively the Exceptions message could be improved).

Reply:
Yeah, NoMethodError is inappropriate for this!

Reply:
Updated PR with just slightly improved error message. Changing the NoMethodError (in `Undecodable` ?) should be a TODO (that I will not tackle).